### PR TITLE
Added function eigvals

### LINKED ISSUE #8806: eigvals

### PR DESCRIPTION
**Pull request by hyadav2k**
Close #8806

### CODE REVIEW

File: ivy/functional/backends/numpy/experimental/linear_algebra.py [hunk 1]
```diff
@@ -108,3 +108,10 @@ def eig(x: np.ndarray, /) -> Tuple[np.ndarray]:
 
 
 eig.support_native_out = False
+
+
+def eivalsg(x: np.ndarray, /) -> np.ndarray:
```
Comment (kurshakuz) [inline ivy/functional/backends/numpy/experimental/linear_algebra.py:113]: Can you please let me know what is `eivalsg` function here? What is the difference from `eivals`?

File: ivy/functional/backends/torch/experimental/linear_algebra.py [hunk 2]
```diff
@@ -119,5 +119,17 @@ def matrix_exp(
...
 
 
-def eig(x: torch.Tensor, /) -> Tuple[torch.Tensor]:
-    return torch.linalg.eig(x)
+def eig(x: torch.Tensor, /) -> Tuple[torch.Tensor, ...]:
+    if not torch.is_complex(x):
+        ret = torch.linalg.eig(x.to(torch.complex128))
+    else:
+        ret = torch.linalg.eig(x)
+    return tuple(ret)
```
Comment (kurshakuz) [inline ivy/functional/backends/torch/experimental/linear_algebra.py:127]: Was the original function implementation throwing errors before your fix? Maybe we can split this PR into two. In the first you implement eigvals and in the other one you fix eig implementation?

**Review by kurshakuz — COMMENTED** [review]
Hey! Thank you for your contribution! I have left a couple of comments here to understand the context better. For now implementation is looking good to me, but I will run tests later. Can you please for now answer to my comments and fix the lint errrors? Thanks!

File: ivy/functional/backends/jax/experimental/linear_algebra.py [hunk 3]
```diff
@@ -102,4 +102,9 @@ def matrix_exp(
 
 
 def eig(x: JaxArray, /) -> Tuple[JaxArray]:
-    return jnp.linalg.eig(x)
+    w, v = jnp.linalg.eig(x)
+    return w, v
```
[the remark this answers is not in the document]
Comment (hyadav2k) [reply ivy/functional/backends/jax/experimental/linear_algebra.py:106]: Sure, it was throwing merge conflicts that's why I changed it. I will undo the changes. Thanks for reviewing.

File: ivy/functional/backends/numpy/experimental/linear_algebra.py (hunk 1 above)
Comment (hyadav2k) [reply ivy/functional/backends/numpy/experimental/linear_algebra.py:113]: It is a typo. Thanks for pointing out. Will fix it.

File: ivy/functional/backends/torch/experimental/linear_algebra.py (hunk 2 above)
Comment (hyadav2k) [reply ivy/functional/backends/torch/experimental/linear_algebra.py:127]: Maybe this was old implementation, that is why I did so. I will try to undo the changes and check once, how it goes. Thanks

**Comment by hyadav2k** [comment]
Hey @kurshakuz, I have resolved the lint errors arising due to my changes. 
There is one issue I am facing, the **eig** for container is not working in my local even though the master is up to date with origin. 
```python
        >>> x = ivy.array([[1,2], [3,4]])
        >>> c = ivy.Container({'x':{'xx':x}})
        >>> c.eig()
```
The o/p is 
```
Traceback (most recent call last):
  File "C:\Program Files\JetBrains\PyCharm 2022.3\plugins\python\helpers\pydev\pydevconsole.py", line 364, in runcode
    coro = func()
           ^^^^^^
  File "<input>", line 1, in <module>
  File "C:\Users\harsh\ivy\ivy\container\linear_algebra.py", line 643, in eig
    return self.static_eig(
           ^^^^^^^^^^^^^^^^
TypeError: ContainerWithLinearAlgebraExperimental.static_eig() got an unexpected keyword argument 'out'
```
**eigvals** is also facing the same issue. 
Kindly look into it. Thanks.

**Comment by kurshakuz** [comment]
@hyadav2k hey! I have also checked this code snippet and it fails due to the same error. I see that your PR does not depend on `eig()` method, so I will for now suggest to focus on implementing the `eigvals()`. The only thing I would ask you is to report this issue in the discord if you can.

Can you then show me the output of the `test_eigvals` test? If it passes all of them, then it is good to go!

**Comment by hyadav2k** [comment]
> The only thing I would ask you is to report this issue in the discord if you can.

Sure, I have reported it in [discord](https://discord.com/channels/799879767196958751/982728618469912627/1058396697123889252).
> Can you then show me the output of the `test_eigvals` test? If it passes all of them, then it is good to go!

Sure, I ran `pytest test_linalg.py::test_eigvals`. It passed all the tests.
```
========================================================================================== test session starts ==========================================================================================
platform linux -- Python 3.8.10, pytest-7.1.2, pluggy-1.0.0
rootdir: /workspaces/ivy
plugins: metadata-2.0.4, typeguard-2.13.3, json-report-1.5.0, hypothesis-6.55.0
collected 4 items                                                                                                                                                                                       

test_linalg.py ....                                                                                                                                                                               [100%]

=========================================================================================== warnings summary ============================================================================================
../../../../../../../usr/local/lib/python3.8/dist-packages/flatbuffers/compat.py:19
  /usr/local/lib/python3.8/dist-packages/flatbuffers/compat.py:19: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import imp

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================================================================================ 4 passed, 1 warning in 74.76s (0:01:14) ================================================================================
```
@kurshakuz, kindly review the latest commit. Thanks.

File: ivy/functional/backends/tensorflow/experimental/linear_algebra.py [hunk 4]
```diff
@@ -59,9 +59,18 @@ def matrix_exp(
 
 
 def eig(
-    x: Union[tf.Tensor],
+    x: Union[tf.Tensor, tf.Variable],
```
Comment (kurshakuz) [inline ivy/functional/backends/tensorflow/experimental/linear_algebra.py:62]: I don't think you need to change this line, as `eig` only expects Tensor as input as stated here: https://www.tensorflow.org/api_docs/python/tf/linalg/eig

File: ivy/functional/backends/tensorflow/experimental/linear_algebra.py [hunk 5]
```diff
@@ -59,9 +59,18 @@ def matrix_exp(
...
     /,
 ) -> Tuple[tf.Tensor]:
     if not ivy.dtype(x) in (ivy.float32, ivy.float64, ivy.complex64, ivy.complex128):
         return tf.linalg.eig(tf.cast(x, tf.float64))
     return tf.linalg.eig(x)
+
+
+def eigvals(
+    x: Union[tf.Tensor, tf.Variable],
```
Comment (kurshakuz) [inline ivy/functional/backends/tensorflow/experimental/linear_algebra.py:71]: Similarly here: https://www.tensorflow.org/api_docs/python/tf/linalg/eigvals

**Review by kurshakuz — COMMENTED** [review]
Hey! I have tested your implementation locally and it passing the test. The only thing I think you would need to change is return the tf.Tensor as the only input type to the `eig` method and remove it from the `eigvals` as it is stated in the TF documentation. Otherwise LGTM

File: ivy/functional/backends/tensorflow/experimental/linear_algebra.py (hunk 4 above)
Comment (hyadav2k) [reply ivy/functional/backends/tensorflow/experimental/linear_algebra.py]: Thanks for reviewing, I made the necessary changes.

**Review by kurshakuz — APPROVED** [review]
LGTM! Thank you for your contribution! 🙂

**Pull request merged**